### PR TITLE
[FEATURE] Tag filter highlight on map page #5313

### LINKED ISSUE #5313: [FEATURE] Tag filter highlight on map page
### Description

The tag based filtering on the map page should stay highlighted on filter selection.  It should mimic the functionality on the search page. 

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by joaoGabriel55**
## Fixes Issue
Closes #5313

## Changes proposed

- [x]  Implement Tag filter highlight on map page

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

image

## Note to reviewers

How can I write map.spec test? I followed the docs but for some reason the test was not executed. Do I need to execute the tests on docker container env?

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/5331"></a>

### CODE REVIEW

File: pages/map.js [hunk 1]
```diff
@@ -55,11 +55,24 @@ export async function getServerSideProps() {
...
   const [filteredUsers, setFilteredUsers] = useState([]);
+  const [selectedTags, setSelectedTags] = useState(new Set());
+
   let results = [];
 
+  const updateSelectedTagsFilter = (tagSelected) => {
+    const updatedSelectedTags = new Set(selectedTags);
+
+    if (updatedSelectedTags.has(tagSelected)) updatedSelectedTags.delete(tagSelected)
+    else updatedSelectedTags.add(tagSelected)
```
Comment (eddiejaoude) [inline pages/map.js:66]: Please use the same coding style as the rest of the project, for example...

```suggestion
    if (updatedSelectedTags.has(tagSelected)) {
       updatedSelectedTags.delete(tagSelected)
     }
    
    updatedSelectedTags.add(tagSelected)
```

**Comment by joaoGabriel55** [comment]
About spec, I saw that my current OS: Zorin OS, cannot install `playwright`, apparently only Ubuntu distro is supported, does anyone already had problems to install `playwright`?

**Comment by eddiejaoude** [comment]
>About map.spec.js test creation, I saw that my current OS: Zorin OS, cannot install playwright, apparently only Ubuntu distro is supported, does anyone already had problems to install playwright?

What is the error you get?

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍

**Comment by eddiejaoude** [comment]
Issue raise for bug when toggling off tag https://github.com/EddieHubCommunity/LinkFree/issues/5420

note: I forgot to write these tests can come in another pull request

**Pull request merged**